lib/portage/dbapi/bintree.py

```
@@ -146,23 +148,33 @@ def aux_get(self, mycpv, wants, myrepo=None):
 		elif not self.bintree._remotepkgs or \
 			not self.bintree.isremote(mycpv):
 			try:
-				tbz2_path = self.bintree._pkg_paths[instance_key]
+				binpkg_path = self.bintree._pkg_paths[instance_key]
 			except KeyError:
 				raise KeyError(mycpv)
-			tbz2_path = os.path.join(self.bintree.pkgdir, tbz2_path)
+			binpkg_path = os.path.join(self.bintree.pkgdir, binpkg_path)
 			try:
-				st = os.lstat(tbz2_path)
+				st = os.lstat(binpkg_path)
 			except OSError:
 				raise KeyError(mycpv)
-			metadata_bytes = portage.xpak.tbz2(tbz2_path).get_data()
+			if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+				metadata_bytes = portage.xpak.tbz2(binpkg_path).get_data()
+			elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+				metadata_bytes = portage.gpkg.gpkg(self.settings, mycpv,
+					binpkg_path).get_metadata()
+			else:
+				return
```

We should raise InvalidBinaryPackageFormat here.